internal/webhooks/clusterclass.go

```diff
@@ -380,12 +380,19 @@ func (webhook *ClusterClass) getClustersUsingClusterClass(ctx context.Context, c
 	clusters := &clusterv1.ClusterList{}
 	err := webhook.Client.List(ctx, clusters,
 		client.MatchingFields{index.ClusterClassNameField: clusterClass.Name},
-		client.InNamespace(clusterClass.Namespace),
```

Comment:
Wondering if we should update the index, so it contains both namespace and name of the cluster class

Reply:
Thinking a little bit more about this, I think we change this index so performance in the webhook are optimized and also the resulting code will be more readable (both here and in clusterClassToCluster)

Reply:
It seems to me that in this method there is not enough information to know cluster namespace due to duality of the namespace reference. 

If an index is added, there will be another `Client.List` call for `client.MatchingFields{index.ClusterClassNamespaceField: clusterClass.Namespace}`, and here I’m not sure how would an empty field be treated - matching an empty string or matching no selector at all?

If the first one the case, then the method will combine two list responses, and it will be more readable. If not, it will only increase the number of calls.

Reply:
Ok, seems it is not an issue, as the index can use `GetClass()` method which resolves ambiguity. Should be added now.